### PR TITLE
Partial serial fix for io_gdb.

### PR DESCRIPTION
**Pull request by vifino**
At least it allows the serial rsocket backend to be called, but I can't quite connect to my BMP, so I am not 100% sure if it works or not, but it shouldn't be this part anymore. Something else might have broken something, however.

### CODE REVIEW

File: libr/io/p/io_gdb.c [hunk 1]
```diff
@@ -85,31 +87,50 @@ static RIODesc *__open(RIO *io, const char *file, int rw, int mode) {
...
+		isdev = true;
+	}
+
+	if (isdev) {
+		port = strrchr (host, '@');
+		if (port) {
+			*port = '\0';
+			port++;
+
+			pid = strrchr (port, ':');
```
Comment (radare) [inline libr/io/p/io_gdb.c]: Strrchr is not portable. Check what we have in until/str

File: libr/io/p/io_gdb.c [hunk 2]
```diff
@@ -85,31 +87,50 @@ static RIODesc *__open(RIO *io, const char *file, int rw, int mode) {
...
+		}
+	} else {
+		if (r_sandbox_enable (0)) {
+			eprintf ("sandbox: Cannot use network\n");
+			return NULL;
+		}
+
+		port = strchr (host , ':');
+		if (!port) {
+			eprintf ("Port not specified. Please use gdb://host:port[/pid] or gdb:///dev/DEVICENAME[@speed][:pid].\n");
```
Comment (radare) [inline libr/io/p/io_gdb.c]: Maybe use multiple lines instead of a long single line to make it more readable

Comment (vifino) [inline libr/io/p/io_gdb.c]: In the output or just in the source?

**Comment by radare** [comment]
Output

**Comment by vifino** [comment]
Are multiple eprintf statements OK? I am not sure how to indent it otherwise.

**Pull request merged**